### ISSUE TITLE
Clarify what's in apk ("FOSS variant" as a future raises the issue of what the other variant is)

### ISSUE THREAD

**Issue by gdt** [ouchadam/small-talk#140]
First thanks for working with @IzzySoft to head towards F-Droid inclusion.  I view being in the main F-Droid repo as more or less a certification that the app (including the apk, not just the source repo) is actually Free Software, and gives important information about what the app actually does (if no tracking antifeatures) or more precisely does not do.

The README lists a feature "FOSS variant".  That implies that there is some other variant that isn't Free Software, but that isn't explained.

Perhaps change to "FOSS build variant (no crashlytics, no Firebase)" which would give the hint that the standard build has these non-Free components.

Separately from that, it would be good to tell people that the app will use those services, and especially to say if crash reports or other data are telemetered in the background or only on crash, and whether it's only if the user opts in.  And, for FCM, what data is sent in the push messages, even though that's probably a matrix thing.   "please check in with server for registration cookie X" is one thing, and anything more is a bit scary.

**Comment by gdt** [comment]
No, I'm not aware of any good examples.  I think this is a widespread issue.

Basically I see two separate things to explain:
  1) In each version of the app, describe anything that would either a) fail the F-Droid inclusion criteria or b) trigger an anti-feature.  Your quick summary is good.
  2) Somehow give a specification for behavior, something like the following

The app will not transmit data to anywhere, other than a) to the homeserver you specify to get/send messages using the matrix protocol, after you have given a homeserver URL and login name and asked to log in, b) to google for crash reports, only after fatal or non-fatal exceptions, and then only after you click yes to an opt-in dialog, c) to register for push notifications with FCM via the included Google Play Services library.   Note that push notifcations for matrix follow MCS foobar which states that [totally making this up] push notification payloads contain only a count of calls, DMs ,and group chat messages not yet retrieved by this device; they never contain message contents or correspondent identity.

**Comment by ouchadam** [comment]
sounds great :100:  is there a preferred location for this information? (readme, wiki, separate doc, in app etc)

**Comment by gdt** [comment]
My view is that describing what an app does, to include what it does not do, is hugely important and should be demanded by all users of all apps, and thus belongs in the top-level README.  For a well-behaved app it is brief.  And readme tends to get copied into f-droid or other store descriptions; it belongs in those too.

Probably there should be in the  app an "about" which shows the readme content, too, but I'm much less concerned about that, than being able to understand what the app says it does before I decide it's ok to install.

**Comment by ouchadam** [comment]
I've added a section to readme, although it's not very brief :thinking:  https://github.com/ouchadam/small-talk/blob/970bd4db57eddea7814e3e6df3a090b2deb66360/README.md#data-and-privacy

**Comment by gdt** [comment]
I realize why you think it's not brief, but you touched on things that on reading I think matter but I hadn't thought to ask about.  Overall I think it's great, and thank you for taking my comment seriously.

For the push bullets in the two  versions, I'd say
```
Push notifications provided by Google Firebase (requires ROM support, Firebase messaging enabled (normally present on ROMs that are not "de-Googled"), and homeserver support for FCM).
Push notifications provided by Unifed Push (requires distributor app and homeserver support for that app).
```
on the theory that both have a library to talk to not-in-app push code, and in either case it might be present or not present.   On e.g. Lineage FCM won't work, and it won't work on CalyxOS if one has disabled Google Device Registration and FCM in microg.
